### PR TITLE
upgrade latest state on transit

### PR DESCRIPTION
**Pull request by vmidyllic**

### CODE REVIEW

File: src/identity/identity-wallet.ts [hunk 1]
```diff
@@ -266,6 +266,18 @@ export interface IIdentityWallet {
...
   getProfileByVerifier(verifier: string): Promise<Profile | undefined>;
+
+  /**
+   *
+   * updates latest identity state in storage with given state or latest from the trees.
+   *
+   * @param {string} issuerDID -  identifier of the verifier
+   * @param {boolean} published - if states is published onchain
+   * @param {string} treeState -  contains state to upgrade
+   * @returns `{Promise<Profile>}`
```
Comment (volodymyr-basiuk) [inline src/identity/identity-wallet.ts:277]: Profile or void ?

File: src/identity/identity-wallet.ts [hunk 2]
```diff
@@ -266,6 +266,18 @@ export interface IIdentityWallet {
...
    */
   getProfileByVerifier(verifier: string): Promise<Profile | undefined>;
+
+  /**
+   *
+   * updates latest identity state in storage with given state or latest from the trees.
+   *
+   * @param {string} issuerDID -  identifier of the verifier
+   * @param {boolean} published - if states is published onchain
+   * @param {string} treeState -  contains state to upgrade
```
Comment (volodymyr-basiuk) [inline src/identity/identity-wallet.ts:276]: param TreeState

File: src/identity/identity-wallet.ts [hunk 3]
```diff
@@ -266,6 +266,18 @@ export interface IIdentityWallet {
    * @returns `{Promise<Profile>}`
    */
   getProfileByVerifier(verifier: string): Promise<Profile | undefined>;
+
+  /**
+   *
+   * updates latest identity state in storage with given state or latest from the trees.
+   *
+   * @param {string} issuerDID -  identifier of the verifier
```
Comment (volodymyr-basiuk) [inline src/identity/identity-wallet.ts:274]: param DID

**Review by volodymyr-basiuk — APPROVED** [review]

File: src/proof/proof-service.ts [hunk 4]
```diff
@@ -282,6 +282,9 @@ export class ProofService implements IProofService {
     const proof = await this._prover.generate(inputs, CircuitId.StateTransition);
 
     const txId = await stateStorage.publishState(proof, ethSigner);
+
+    this._identityWallet.updateIdentityState(did, true, newTreeState);
```
Comment (volodymyr-basiuk) [inline src/proof/proof-service.ts:286]: await here ?

Comment (vladyslavmunin) [reply src/proof/proof-service.ts]: fixed

File: src/identity/identity-wallet.ts (hunk 3 above)
Comment (vladyslavmunin) [reply src/identity/identity-wallet.ts]: fixed

File: src/identity/identity-wallet.ts (hunk 2 above)
Comment (vladyslavmunin) [reply src/identity/identity-wallet.ts]: fixed

File: src/identity/identity-wallet.ts (hunk 1 above)
Comment (vladyslavmunin) [reply src/identity/identity-wallet.ts]: fixed

**Pull request merged**